newIDE/app/src/EventsSheet/EventsTree/Instruction.js

```diff
@@ -146,14 +166,50 @@ const Instruction = (props: Props) => {
           }
 
           const parameterMetadata = metadata.getParameter(parameterIndex);
-          const parameterType = parameterMetadata.getType();
+          const parameterSubType = parameterMetadata.getType();
+          const parameterType = gd.ParameterMetadata.isExpression(
+            'number',
+            parameterSubType
+          )
+            ? 'number'
+            : gd.ParameterMetadata.isExpression('string', parameterSubType)
+            ? 'string'
+            : parameterSubType;
+          let expressionIsValid = true;
+          if (
+            parameterType === 'number' ||
+            parameterType === 'string' ||
+            gd.ParameterMetadata.isExpression('variable', parameterType)
+          ) {
+            const parser = new gd.ExpressionParser2(
+              gd.JsPlatform.get(),
+              globalObjectsContainer,
+              objectsContainer
+            );
+            const expressionNode = instruction
+              .getParameter(parameterIndex)
+              .getRootNode(parameterType, parser)
+              .get();
+            const expressionValidator = new gd.ExpressionValidator();
+            expressionNode.visit(expressionValidator);
+            expressionIsValid = expressionValidator.getErrors().size() === 0;
```

Comment:
Maybe this should be cached too?

Reply:
I'm not sure what the cost is exactly, so hard to say. Ideally we want to avoid caching, but it also means that for now we're running validation at every render, which can be time consuming.

Add some performance.now() to see what is the time to run this validation on average? And on a very large expression (1000 characters?)